PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = 'hr.employee'
-    _inherit = 'hr.employee.public'
-    _inherit = 'fleet.vehicle'
-    _inherit = 'fleet.vehicle.assignation.log'
-    _inherit = 'fleet.vehicle.log.contract'
-    _inherit = 'fleet.vehicle.log.services'
-    _inherit = 'fleet.vehicle.odometer'
-    _inherit = 'mail.activity.plan.template'

Total Changes: 23

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_fleet/models/employee.py
+++ b/addons/hr_fleet/models/employee.py
@@ -5,8 +5,8 @@ from odoo import _, api, fields, models
 from odoo.exceptions import ValidationError
 
 
-class Employee(models.Model):
-    _inherit = 'hr.employee'
+class HrEmployee(models.Model):
+    _inherit = ['hr.employee']
 
     employee_cars_count = fields.Integer(compute="_compute_employee_cars_count", string="Cars", groups="fleet.fleet_group_manager")
     car_ids = fields.One2many(
@@ -86,7 +86,7 @@ class Employee(models.Model):
                     car.driver_id = user.partner_id
 
 
-class EmployeePublic(models.Model):
-    _inherit = 'hr.employee.public'
+class HrEmployeePublic(models.Model):
+    _inherit = ['hr.employee.public']
 
     mobility_card = fields.Char(readonly=True)

--- a/addons/hr_fleet/models/fleet_vehicle.py
+++ b/addons/hr_fleet/models/fleet_vehicle.py
@@ -4,8 +4,9 @@
 from odoo import _, api, fields, models
 from odoo.exceptions import ValidationError
 
+
 class FleetVehicle(models.Model):
-    _inherit = 'fleet.vehicle'
+    _inherit = ['fleet.vehicle']
 
     mobility_card = fields.Char(compute='_compute_mobility_card', store=True)
     driver_employee_id = fields.Many2one(

--- a/addons/hr_fleet/models/fleet_vehicle_assignation_log.py
+++ b/addons/hr_fleet/models/fleet_vehicle_assignation_log.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class FleetVehicleAssignationLog(models.Model):
-    _inherit = 'fleet.vehicle.assignation.log'
+    _inherit = ['fleet.vehicle.assignation.log']
 
     driver_employee_id = fields.Many2one('hr.employee', string='Driver (Employee)', compute='_compute_driver_employee_id', store=True, readonly=False)
     attachment_number = fields.Integer('Number of Attachments', compute='_compute_attachment_number')

--- a/addons/hr_fleet/models/fleet_vehicle_log_contract.py
+++ b/addons/hr_fleet/models/fleet_vehicle_log_contract.py
@@ -3,8 +3,9 @@
 
 from odoo import fields, models, _
 
+
 class FleetVehicleLogContract(models.Model):
-    _inherit = 'fleet.vehicle.log.contract'
+    _inherit = ['fleet.vehicle.log.contract']
 
     purchaser_employee_id = fields.Many2one(
         related='vehicle_id.driver_employee_id',

--- a/addons/hr_fleet/models/fleet_vehicle_log_services.py
+++ b/addons/hr_fleet/models/fleet_vehicle_log_services.py
@@ -3,8 +3,9 @@
 
 from odoo import api, fields, models
 
+
 class FleetVehicleLogServices(models.Model):
-    _inherit = 'fleet.vehicle.log.services'
+    _inherit = ['fleet.vehicle.log.services']
 
     purchaser_employee_id = fields.Many2one(
         'hr.employee', string="Driver (Employee)",

--- a/addons/hr_fleet/models/fleet_vehicle_odometer.py
+++ b/addons/hr_fleet/models/fleet_vehicle_odometer.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class FleetVehicleOdometer(models.Model):
-    _inherit = 'fleet.vehicle.odometer'
+    _inherit = ['fleet.vehicle.odometer']
 
     driver_employee_id = fields.Many2one(
         related='vehicle_id.driver_employee_id', string='Driver (Employee)',

--- a/addons/hr_fleet/models/mail_activity_plan_template.py
+++ b/addons/hr_fleet/models/mail_activity_plan_template.py
@@ -6,7 +6,7 @@ from odoo import exceptions
 
 
 class MailActivityPlanTemplate(models.Model):
-    _inherit = 'mail.activity.plan.template'
+    _inherit = ['mail.activity.plan.template']
 
     responsible_type = fields.Selection(
         selection_add=[('fleet_manager', "Fleet Manager")],
